scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -18,11 +18,15 @@
 class HttpCompressionMiddleware(object):
     """This middleware allows compressed (gzip, deflate) traffic to be
     sent/received from web sites"""
+
+    def __init__(self, settings):
+        self.keep_encoding_header = settings.getbool('HTTPCOMPRESSION_HEADERS_KEEP')
```

Comment:
What about renaming `HTTPCOMPRESSION_HEADERS_KEEP` to `COMPRESSION_KEEP_ENCODING_HEADER`?

Although `HTTPCOMPRESSION` seems in line with other middlewares, this middleware already uses `COMPRESSION` as prefix (there’s a `COMPRESSION_ENABLED` setting to enable the middleware).

And since this setting only affects the `Accept-Encoding` header, I think it makes sense to make the setting name explicit about that.

Reply:
Sounds good. I wasn't sure what to name it so tried to keep it close to where it would be making an effect.